### PR TITLE
chore(client): patch for father sub edge

### PR DESCRIPTION
**Pull request by Thespica**
## Purpose of the PR

1. add a edge label type called `GENERAL`
2. add unit test for edge label type

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [ ]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/builder/EdgeBuilder.java [hunk 1]
```diff
@@ -55,10 +56,19 @@ public EdgeBuilder(LoadContext context, InputStruct struct,
         super(context, struct);
         this.mapping = mapping;
         this.edgeLabel = this.getEdgeLabel(this.mapping.label());
-        this.sourceLabel = this.getVertexLabel(this.edgeLabel.sourceLabel());
-        this.targetLabel = this.getVertexLabel(this.edgeLabel.targetLabel());
         this.nonNullKeys = this.nonNullableKeys(this.edgeLabel);
-        // Ensure that the source/target id fields are matched with id strategy
+        if (this.edgeLabel.edgeLabelType().general()) {
+            // If creating a general type edge, the loader cannot obtain the vertexLabel information of both ends
```
Comment (imbajin) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/builder/EdgeBuilder.java:61]:
```suggestion
            // If create a general type edge, the loader can't obtain the vertexlabel info of both ends
```

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/builder/EdgeBuilder.java [hunk 2]
```diff
@@ -189,7 +199,7 @@ private void checkIdFields(VertexLabel vertexLabel, List<String> fields) {
         } else if (vertexLabel.idStrategy().isPrimaryKey()) {
             E.checkArgument(fields.size() >= 1,
                             "The source/target field must contains some " +
-                            "columns when id strategy is PrimaryKey");
+                            "columns when id strategy is CUSTOMIZE");
```
Comment (imbajin) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/builder/EdgeBuilder.java:202]: logic confused?

**Review by imbajin — DISMISSED** [review]

File: hugegraph-client/src/main/java/org/apache/hugegraph/structure/constant/EdgeLabelType.java [hunk 3]
```diff
@@ -46,6 +48,10 @@ public boolean normal() {
         return this == NORMAL;
     }
 
+    public boolean general(){
```
Comment (VGalaxies) [inline hugegraph-client/src/main/java/org/apache/hugegraph/structure/constant/EdgeLabelType.java:51]:
```suggestion
    public boolean general() {
```

**Review by VGalaxies — APPROVED** [review]

**Review by imbajin — APPROVED** [review]
